py/objtype.c

```diff
@@ -652,11 +663,15 @@ STATIC void mp_obj_instance_load_attr(mp_obj_t self_in, qstr attr, mp_obj_t *des
 STATIC bool mp_obj_instance_store_attr(mp_obj_t self_in, qstr attr, mp_obj_t value) {
     mp_obj_instance_t *self = MP_OBJ_TO_PTR(self_in);
 
+    if (!(self->base.type->flags & TYPE_FLAG_HAS_SPECIAL_ACCESSORS)) {
```

Comment:
Why this overhead code isn't wrapped with `#if ENABLE_SPECIAL_ACCESSORS`?

Reply:
Same as above, the compiler is smart enough to eliminate it.  And wrapping this with `ENABLE_SPECIAL_ACCESSORS` then requires wrapping the label `skip_special_accessors` to avoid compiler warnings, which makes the code more ugly / harder to read.